prefect_aws/s3.py

```diff
@@ -337,9 +339,118 @@ def _get_s3_client(self) -> boto3.client:
             s3_client = self.aws_credentials.get_boto3_session().client(
                 service_name="s3"
             )
-
+        else:
+            raise ValueError(
+                "S3 Bucket requires either a minio_credentials"
+                "field or an aws_credentials field."
+            )
         return s3_client
 
+    def _get_bucket_resource(self) -> boto3.resource:
+        if self.minio_credentials:
+            bucket = (
+                self.minio_credentials.get_boto3_session()
+                .resource("s3", endpoint_url=self.endpoint_url)
+                .Bucket(self.bucket_name)
+            )
+
+        elif self.aws_credentials:
+            bucket = (
+                self.aws_credentials.get_boto3_session()
+                .resource("s3")
+                .Bucket(self.bucket_name)
+            )
+        else:
+            raise ValueError(
+                "S3 Bucket requires either a minio_credentials"
+                "field or an aws_credentials field."
+            )
+        return bucket
+
+    async def get_directory(
+        self, from_path: Optional[str] = None, local_path: Optional[str] = None
+    ) -> None:
+        """
+        Copies a folder from the configured S3 bucket to a local directory.
+
+        Defaults to copying the entire contents of the block's basepath to the current
+        working directory.
+
+        Args:
+            from_path: Path in S3 bucket to download from. Defaults to the block's
+                configured basepath.
+            local_path: Local path to download S3 contents to. Defaults to the current
+                working directory.
+        """
+        if from_path is None:
+            from_path = str(self.basepath) if self.basepath else ""
```

Comment:
I think basepath is already a `str` in `cast_pathlib`

Reply:
I realized that basepath's original purpose is for `s3://` or `gs://`. Are those no longer needed? If still need, not sure if the logic below would work.

Reply:
The prefixes are no longer needed because we're only working with AWS S3 via boto3 instead of using fsspec.